spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -383,6 +383,22 @@ def defined_private_method
         object.implemented
       end
 
+      it 'wraps proxies in a reentrant mutex so we dont trigger deadlocks' do
```

Comment:
I'm only using `RSpec::Mocks.space.proxy_for ` here because normal mocks usage triggers the thread deadlock, but then triggers an infinite loop anyway so i needed something more direct.